src/seismometer/table/analytics_table.py

```diff
@@ -0,0 +1,715 @@
+        metric_values: List[float] = [0.1, 0.2],
+        metrics_to_display: Optional[List[str]] = None,
+        title: str = "Model Performance Statistics",
+        top_level: str = "Score",
+        table_config: AnalyticsTableConfig = AnalyticsTableConfig(),
+        statistics_data: Optional[pd.DataFrame] = None,
+        per_context: bool = False,
+    ):
+        """
+        Initializes the PerformanceMetrics object with the necessary data and parameters.
+
+        Parameters
+        ----------
+        df: Optional[pd.DataFrame]
+            The DataFrame containing the model's predictions and targets, by default None.
+        score_columns: Optional[List[str]]
+            A list of column names corresponding to model prediction scores to be evaluated, by default None.
+        target_columns: Optional[List[str]]
+            A list of column names corresponding to targets, by default None.
+        metric: str
+            Performance metrics will be presented for the provided values of this metric.
+        metric_values: List[float]
+            Values for the specified metric to derive detailed performance statistics, by default [0.1, 0.2].
+        metrics_to_display: Optional[List[str]]
+            List of metrics to include in the table, by default None. The default behavior is to shows all columns
+            in GENERATED_COLUMNS, which is a dictionary mapping metric names to their corresponding column names.
+        title: str
+            The title for the performance statistics report, by default "Model Performance Statistics".
+        top_level: str
+            The primary grouping category in the performance report, by default 'Score'.
+        decimals: int
+            The number of decimal places for rounding numerical results, by default 3.
+        columns_show_percentages: Union[str, List[str]]
+            Columns that will display values as percentages in the report, by default "Prevalence".
+        columns_show_bar: Optional[Dict[str, str]]
+            Columns mapped to specific colors for bar representation in the report, by default None.
+        percentages_decimals: int
+            The number of decimal places for percentage values in the report, by default 0.
+        statistics_data: Optional[pd.DataFrame]
+            Additional performance metrics statistics, will be joined with the statistics data generated
+            by the code, by default None.
+
+        Raises
+        ------
+        ValueError
+            If any of the provided spanner colors are not recognized as valid color names.
+            If neither "df" nor "statistics_data" is provided.
+            If "df" is provided but either "score_columns" or "target_columns" is not provided.
+            If the provided metric name is not recognized.
+            If the provided top_level name is not "Score" or "Target".
+        """
+        sg = Seismogram()
+        self.df = df if df is not None else sg.dataframe
```

Comment:
why do we want to pass in a dataframe here if we are going to set it by default?

Reply:
removed df.